scripts/lib/CIME/SystemTests/eri.py

```diff
@@ -112,11 +112,13 @@ def run_phase(self):
 
         dout_sr1 = clone1.get_value("DOUT_S_ROOT")
 
-        # force cam namelist to write out initial file at end of run
-        if os.path.exists("user_nl_cam"):
-            if "inithist" not in open("user_nl_cam", "r").read():
-                with open("user_nl_cam", "a") as fd:
-                    fd.write("inithist = 'ENDOFRUN'\n")
+        # force cam/eam namelist to write out initial file at end of run
+        for model in ["cam", "eam"]:
+          user_nl = "user_nl_{}".format(model)
```

Comment:
Just for informational purposes (doesn't need to be fixed for this PR): this doesn't handle multi-instance cases. There is a more general function here https://github.com/esmci/cime/blob/master/scripts/lib/CIME/SystemTests/test_utils/user_nl_utils.py#L8 that handles multi-instance, but that function would need to be extended if you want to replicate all of the logic here, including something like an optional argument to return quietly if the given file doesn't exist.